fconfig/config.go

```diff
@@ -0,0 +1,105 @@
+package fconfig
+
+import (
+	"os"
+	"regexp"
+
+	"github.com/Flahmingo-Investments/helpers-go/ferrors"
+	"github.com/Flahmingo-Investments/helpers-go/gcpauth"
+	"github.com/joho/godotenv"
+	"github.com/spf13/viper"
+)
+
+var secretRegex = regexp.MustCompile(`^gSecret://(?P<Path>.+)`)
+var envFileRegex = regexp.MustCompile(`\.env$`)
+
+// LoadConfigViper loads the configuration from a given file.
+// For environment variables use the following format int the FILE. `${<Env var>}`
+// For Google secrets, use the following format.
+// gSecret://projects/<projectId>/secrets/<secretName>/versions/<version id or 'latest'>
+// In order to parse the config, run v.Unmarshal(<config struct>), and ensure all your properties are exported/public
+func LoadConfigViper(files []string) (*viper.Viper, error) {
+	var err error
+
+	v := viper.New()
+	v.SetEnvPrefix("")
+	v.AutomaticEnv()
+
+	firstFile := true
+	for _, file := range files {
+		if envFileRegex.MatchString(file) {
+			err = loadEnv(file)
+			if err != nil {
+				return nil, err
+			}
+		} else if firstFile {
+			v.SetConfigFile(file)
+			firstFile = false
+		} else {
+			v.AddConfigPath(file)
+		}
```

Comment:
A single viper instance can load only a single file type.
So, I think, if we want load the `.env` files it is better to do it in `LoadConfig`.

Also, `v.AddConfigPath` add search path to look for the config file so, it totally unnecessary until we have a default config name and type.

Reply:
Would probably be easier to enforce the need for ONE config file, if we don't want to have a program read multiple config files.
That includes removing env vars and such

Reply:
Now we have 1 config file, and if you want to use environment variables you have a `.env` file in the project dir and the program will import it, and you use the `${envVar}` syntax to grab it in the config file.